plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -53,21 +55,23 @@ type pusher struct {
 
 	initNonBlockingChOnce sync.Once
 	startNonBlockCh       chan struct{}
+	retentionSet          map[string]int
 }
 
-func NewPusher(target Target, service CloudWatchLogsService, flushTimeout time.Duration, retryDuration time.Duration, logger telegraf.Logger) *pusher {
+func NewPusher(target Target, service CloudWatchLogsService, flushTimeout time.Duration, retryDuration time.Duration, logger telegraf.Logger, retention int) *pusher {
 	p := &pusher{
-		Target:        target,
-		Service:       service,
-		FlushTimeout:  flushTimeout,
-		RetryDuration: retryDuration,
-		Log:           logger,
-
+		Target:          target,
+		Service:         service,
+		FlushTimeout:    flushTimeout,
+		RetryDuration:   retryDuration,
+		Log:             logger,
+		Retention:       retention,
 		events:          make([]*cloudwatchlogs.InputLogEvent, 0, 10),
 		eventsCh:        make(chan logs.LogEvent, 100),
 		flushTimer:      time.NewTimer(flushTimeout),
 		stop:            make(chan struct{}),
 		startNonBlockCh: make(chan struct{}),
+		retentionSet:    map[string]int{},
```

Comment:
Should this be rename to retentionMap or maybe more descriptive LogGroupRetentionMap?